libs/error-stack/src/report.rs

```diff
@@ -732,8 +730,8 @@ impl<Context> Extend<Self> for Report<Context> {
 }
 
 #[cfg(feature = "wasm")]
-impl<Context> From<Report<Context>> for JsValue {
-    fn from(report: Report<Context>) -> Self {
-        JsValue::from_str(&report.to_string())
+impl<C: Context> From<Report<C>> for JsValue {
+    fn from(report: Report<C>) -> Self {
+        serde_wasm_bindgen::to_value(&report).expect("Failed to serialize report")
```

Comment:
> What about using serde-wasm-bindgen directly in the From trait implementation? Implementing such a trait would be useful in the case where a library using error-stack is meant to be used both within a Rust application and as a wasm package. In this case, returning JsValue isn't really feasible, as it would be nice only when compiling for wasm targets

Threading a reply here. This is definitely a lot better than the previous approach, but I think we're still apprehensive about adding this due to the implications for package bloat.

In my mind there are two aspects here
- Using `error-stack` in wasm contexts (with `wasm-bindgen`) is possible _without_ us adding native support for it in the library here. 
  - You're able to just call `serde_wasm_bindgen::to_value` yourself on the `Report` and create a new type (or just change your wasm function signatures to `JsValue` and not even have a new type).
- Adding support for something like `wasm-bindgen` then opens questions about what other libraries we might want to support.
  - It's true that we do have some features related to external libraries already (`eyre` and `anyhow` for example), but those are more about compatibility layers between _error_ libraries and are used to _create_ `Report`s rather than _consuming_ them.
  - If we do want to add `wasm-bindgen`, then do we also want to add support for `axum`, or `pyo3`, or `actix-web`? We're not convinced we do want to do that (we're also not convinced we don't want to), but we do have limited resources and maintaining native support for these things is not going to come without investment. 

I think for now, we'd prefer not to add this. I believe we're centering on the general philosophy of "we'll extend `error-stack` with things _if people can't do it themselves_.